### ISSUE TITLE
Transparency not working on Windows 10

### ISSUE THREAD

**Issue by cvandal** [zeit/hyper#1638]
- [x ] I am on the [latest](https://github.com/zeit/hyper/releases/latest) Hyper.app version
- [x ] I have searched the [issues](https://github.com/zeit/hyper/issues) of this repo and believe that this is not a duplicate

- **Windows 10**: 
- **1.3.1**: 
- **The issue is reproducible in vanilla Hyper.app**: 

## Issue

I am unable to get transparency working on Hyper 1.31 on Windows 10. I have tried a few plugins, none of which worked, and changing my backgroundColor to `backgroundColor: 'rgba(0,0,0,0.5)'`. Is there anything else that I can try? My `hyper.js` is currently back to default config.

**Comment by Rawnly** [comment]
Try to see this #1625

**Comment by byronpolley** [comment]
+1 Please explain how to enable transparency using flags if it is possible at all. This is my favourite terminal emulator by far and I would like to keep using it but need transparency.

**Comment by Rawnly** [comment]
As sad @Zx-EvM

> Apparently hyper.is team did not implement support for opacity/transparent/vibrancy on Windows.

Please for more infos comment on the [other thread](issue)

[issue]: https://github.com/zeit/hyper/issues/1625

**Comment by cvandal** [comment]
If you're feeling adventurous, I have found that you can enable transparency in Hyper 1.3.1 on Windows 10 by setting `frame: process.platform === 'darwin',` to `frame: false,` and `transparent: process.platform === 'darwin',` to `transparent: true,`. Looks like they only wanted this to work on OSX, and probably for a good reason, so use at your own risk I guess.

**Comment by Zx-EvM** [comment]
@cvandal thanks. it's really working

![screenshot 208](https://cloud.githubusercontent.com/assets/20873402/23831409/a1c893d0-06dd-11e7-8bfb-ad2e842d37ec.png)

**Comment by cvandal** [comment]
I haven't noticed an increase in CPU or MEM consumption, however after enabling transparency I have found that it certainly feels sluggish while entering commands. I can live without transparent terminal, I can't live with a slow terminal so I've reverted my changes :).

**Comment by damien122** [comment]
This is only because if you enable the transparency on windows and then use the View > Full Reload, then the hole window of hyper is completely transparent. I can live with this because I don't use this "Full Reload". I just restart the hole hyper after plugin installations. But I WANT a half transparent window ^^.
BTW, thanks for the workaround @cvandal 👍

**Comment by damien122** [comment]
@stefanivic yes I also found this out today :)
This is the reason why I revert the workaround back to default.

**Comment by albinekb** [comment]
Looks like it's not going to be supported 😞  https://github.com/electron/electron/issues/8532#issuecomment-300715940

**Comment by Terrabits** [comment]
I have enabled @cvandal's hack on Windows 10 without the `Full Reload` problem mentioned by @damien122. I do observe @stefanivic's problem with snapping.

I edited the contents of `app.asar` after installing from the official installer. Unfortunately that means that when the application auto-updates I lose my transparency.

Default electron has a command line flag (`--enable-transparent-visuals`) that does this. Hyper could implement a similar flag (maybe `--transparent` or `--force-transparent`) or add a `transparent` setting to `config` in `.hyper.js`.

**Comment by ryanproud** [comment]
Is the workaround no longer possible?  I'm on version 1.4.8 and not able to find any of the referenced code in app.asar:index.js.

**Comment by johnRivs** [comment]
Hey, any updates on this for Hyper 2?

**Comment by PrestonII** [comment]
@johnRivs @ryanproud - apparently it's a Chromium issue, per the issue referenced by @albinekb : https://github.com/electron/electron/issues/8532#issuecomment-300715940

I don't think they (Hyper) can fix it until Chromium does so on their end...

**Comment by lucleray** [comment]
This can be solved using `browserWindow.setOpacity` ([electron doc](https://electronjs.org/docs/api/browser-window#winsetopacityopacity-windows-macos)). It is compatible with Windows and MacOS.

I made a plugin for that : [hyper-opacity](https://github.com/lucleray/hyper-opacity) 🙂

**Comment by johnRivs** [comment]
@lucleray You know what, it's not the same but it's close enough. I'm gonna check it out.

**Comment by lucleray** [comment]
@johnRivs It's true that it doesn't only set the opacity for the background 🤔

**Comment by kuttor** [comment]
@lucleray THANK YOU so much, this is absolutely amazing -- however, quick question, I can not get BLUR or Vibracvy to work.... just transparency. Is that possible. OMG, I that is possible, then This terminal is absolutely what I been hoping for.

**Comment by chabou** [comment]
@Stanzilla If I remember correctly, opacity was not a core feature due to some bug with mouse interactions. I'll try to find concerned issues.

**Comment by chabou** [comment]
@Stanzilla @lucleray This is the PR that disabled explicitly transparency on Windows: https://github.com/zeit/hyper/pull/1307

I don't know if this is still relevant.

**Comment by lucleray** [comment]
@chabou 

According to the [electron docs](https://electronjs.org/docs/api/frameless-window#transparent-window) :

- `transparent` works on MacOS
- `transparent` will only work when DWM is enabled on Windows (which can be checked like [explained here](https://github.com/electron/electron-docs/blob/master/test/fixtures/system-preferences.md#systempreferencesisaeroglassenabled-windows))
- `transparent` will not work on Linux unless it is started with `--enable-transparent-visuals --disable-gpu`

So, for Windows, instead of explicitly disabling transparency (https://github.com/zeit/hyper/pull/1307), it's possible to enable it if DWM is enabled.

**Comment by albinekb** [comment]
I think it's better suited for a plugin as this will most likely not work 100% all the time and we need to maintain it and also check that it works every time we upgrade electron.

**Comment by pd93** [comment]
@albinekb I agree that this should be a plugin for now while it's a workaround, but if the electron issue ever gets solved conclusively, I feel it should be moved into hyper again to align with the features available on OSX.

There should also be a note somewhere in the README/docs to clarify this position, as lots of people are probably looking for this feature.

**Comment by curpin** [comment]
Has anyone got this working guys? I am using the latest 3.0.2 and still not able to get this transparency working?

**Comment by TylerSeanRau** [comment]
Any update on this?

**Comment by TylerSeanRau** [comment]
I've managed to hack something together with https://www.npmjs.com/package/electron-acrylic-window

![image](https://user-images.githubusercontent.com/5145919/120883560-751b5a00-c5a3-11eb-98dc-1b158a4570ff.png)

I'll post a short how-to-use for those willing to build from source in a bit

**Comment by TylerSeanRau** [comment]
I've added an MR to track the rest of the progress: https://github.com/vercel/hyper/pull/5632
